hal.md

```diff
@@ -0,0 +1,117 @@
+How to porting TA-endpoint on your device
```

Comment:
The subject should be covering HAL.

Reply:
You need to address motivation, concepts, feasible terms, etc.

Reply:
Should the hal.md changed to HAL.md?

Reply:
go ahead.

Reply:
Use fully-qualified subject as well.